### PR TITLE
Enable SmartcarShield (and actual Win32 runtime compat in general)

### LINKED ISSUE #83: External libraries support at runtime
Officially handle `smartcar_shield` across all of SMCE's supported platforms
Let users add their own libraries to the board config and passthrough that to the CMake invocation

Acceptance criteria: works as expected on all three major platforms and is well documented

### PR DESCRIPTION
**Pull request by AeroStun**
Closes #83

### CODE REVIEW

**Comment by AeroStun** [comment]
[[[ For possible post-merge readers ]]]
Arduino global variables are trampoline'd across the DLL boundary because I could not get the MSVC linker to produce an import library that contains the exported data symbol, even though it was marked `__declspec(dllexport)`.
If somehow you know how to do that, please remove the trampolines, fix the symbol, and PR this project

File: CMakeLists.txt [hunk 1]
```diff
@@ -59,19 +59,21 @@ include (BoostDLL)
...
 add_custom_command (TARGET ArduinoRuntime POST_BUILD
-        COMMAND ${CMAKE_COMMAND} -E copy "$<TARGET_FILE:ArduinoRuntime>" "${CMAKE_BINARY_DIR}/share/smce/toolchain/lib/")
+        COMMAND ${CMAKE_COMMAND} -E copy "$<TARGET_FILE:ArduinoRuntime>" "${CMAKE_BINARY_DIR}/share/smce/toolchain/lib/"
+        COMMAND ${CMAKE_COMMAND} -E copy "$<TARGET_LINKER_FILE:ArduinoRuntime>" "${CMAKE_BINARY_DIR}/share/smce/toolchain/lib/"
+        COMMAND ${CMAKE_COMMAND} -E make_directory "${CMAKE_BINARY_DIR}/share/smce/toolchain/libraries"
+        COMMAND ${CMAKE_COMMAND} -E copy_directory "${CMAKE_SOURCE_DIR}/thirdparty/smartcar_shield" "${CMAKE_BINARY_DIR}/share/smce/toolchain/libraries/smartcar_shield")
 file (GENERATE
         OUTPUT "${CMAKE_BINARY_DIR}/share/smce/toolchain/Modules/ArduinoRuntimeLocation.cmake"
-        CONTENT "set (ARDUINO_RUNTIME_FILENAME $<TARGET_FILE_NAME:ArduinoRuntime>)")
+        CONTENT "set (ARDUINO_RUNTIME_FILENAME $<TARGET_FILE_NAME:ArduinoRuntime>)\nset (ARDUINO_RUNTIME_LINK_FILENAME $<TARGET_LINKER_FILE_NAME:ArduinoRuntime>)")
```
Comment (RuthgerD) [inline CMakeLists.txt:76]: `\n` think new line messed up here

Comment (AeroStun) [reply CMakeLists.txt:76]: Works as intended

File: include/ardpolyfills/Wire.h [hunk 2]
```diff
@@ -27,8 +28,8 @@ class TwoWire : public Stream {
     void begin(std::uint8_t with_address);
     void begin(int sda, int scl);
     void begin(int sda, int scl, std::uint8_t with_address);
-    std::size_t requestFrom(std::uint8_t slave_address, std::size_t quantity, bool stop = true);
-    std::uint8_t requestFrom(std::uint8_t slave_address, std::uint8_t quantity, std::uint8_t);
+    std::size_t requestFrom(std::uint8_t slave_address, std::size_t quantity, bool stop = false);
+    std::uint8_t requestFrom(std::uint8_t slave_address, std::uint8_t quantity, std::uint8_t = 0);
```
Comment (crz11111) [inline include/ardpolyfills/Wire.h:32]: Any reason for changing this one?

File: include/ardpolyfills/HardwareSerial.h [hunk 3]
```diff
@@ -73,7 +74,18 @@ struct HardwareSerial : Stream {
...
+#else
+SMCE__DLL_RT_API HardwareSerial& SMCE__DATA_TRAMPOLINE_Serial() noexcept;
+#   ifdef SMCE__COMPILING_USERCODE
+#       define Serial SMCE__DATA_TRAMPOLINE_Serial()
+#   else
+extern HardwareSerial Serial;
+#   endif
+#endif
+
+
```
Comment (davidos-b) [inline include/ardpolyfills/HardwareSerial.h:88]: Remove one line?

**Pull request merged**